src/pages/Loans/LoansOverview/components/LoanModal/BorrowForm.tsx

```diff
@@ -39,72 +63,108 @@ const BorrowForm = ({ asset, variant, position }: BorrowFormProps): JSX.Element
+  });
+
+  const amountFieldName = variant === 'borrow' ? BORROW_AMOUNT : REPAY_AMOUNT;
+  const amount = watch(amountFieldName) || 0;
+  const monetaryAmount = newMonetaryAmount(amount, asset.currency);
+  const newBorrowLimit = getNewBorrowLimitUSDValue(variant, asset.currency, monetaryAmount) || Big(0);
+
+  const isBtnDisabled = !isValidForm(errors) || !isDirty;
+
+  const handleSubmit = (data: BorrowFormData) => {
+    try {
+      console.log(data);
+      refetch();
+    } catch (e) {
+      console.log(e);
+    }
   };
 
   return (
-    <Stack spacing='double'>
-      <div>
-        <H3 id={titleId}>
-          {content.title} {asset.currency.name}
-        </H3>
-        <P>Lorem Ipsum is simply dummy text of the printing and typesetting industry.</P>
-      </div>
-      <Stack>
-        <StyledDItem>
-          <dt>
-            {variant === 'borrow' ? 'Available' : 'Borrowed'} {asset.currency.ticker}:
-          </dt>
-          <dd>
-            <Strong>{formatNumber(variant === 'borrow' ? maximumBorrowableAmount : borrowedAmount)}</Strong> (
-            {formatUSD((variant === 'borrow' ? maximumBorrowableAmount : borrowedAmount) * assetPrice)})
-          </dd>
-        </StyledDItem>
-        <TokenInput
-          onChange={handleInputAmountChange}
-          valueInUSD='$0.00' // TODO: add price computation once RHF is added
-          tokenSymbol={asset.currency.ticker}
-        />
-        <StyledDl>
+    <form onSubmit={h(handleSubmit)}>
+      <Stack spacing='double'>
+        <div>
+          <H3 id={titleId}>
+            {content.title} {asset.currency.name}
+          </H3>
+          <P>Lorem Ipsum is simply dummy text of the printing and typesetting industry.</P>
+        </div>
+        <Stack>
           <StyledDItem>
-            <dt>APY</dt>
-            <dd>{formatNumber(asset.borrowApy.toNumber())}%</dd>
-          </StyledDItem>
-          {variant === 'borrow' && asset.borrowReward && (
-            <StyledDItem>
-              <dt>{asset.borrowReward.currency.ticker} Rewards</dt>
-              <dd>{formatNumber(asset.borrowReward.apy.toNumber())}%</dd>
-            </StyledDItem>
-          )}
-          <StyledDItem>
-            <dt>Borrow Limit</dt>
+            <dt>
+              {variant === 'borrow' ? 'Available' : 'Borrowed'} {asset.currency.ticker}:
+            </dt>
             <dd>
-              {formatUSD(borrowLimitUSDValue?.toNumber() || 0)} -&gt; {formatUSD(newBorrowLimit.toNumber())}
+              <Strong>{formatNumber(variant === 'borrow' ? maximumBorrowable : borrowedAmount)}</Strong> (
+              {formatUSD((variant === 'borrow' ? maximumBorrowable : borrowedAmount) * assetPrice)})
             </dd>
           </StyledDItem>
-        </StyledDl>
-        <CTA onClick={handleFormSubmission} size='large'>
-          {content.title}
-        </CTA>
+          <TokenInput
+            valueInUSD='$0.00' // TODO: add price computation once RHF is added
```

Comment:
Here (and also in its respective place in LendForm) I wanted to add price computation once we have RHF added and can watch the input field, will you add it there within this PR or should I add this to our outstanding work list?

Reply:
I will take care of this in a different PR